pkg/policy/ebpf.go

```diff
@@ -0,0 +1,726 @@
+		}
+		for pid := range procTreeEqualities.Equal {
+			eq := fEqs.procTreeEqualities[uint64(pid)]
+			// Equal == 1, so set n bitmap bit
+			utils.SetBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.procTreeEqualities[uint64(pid)] = eq
+		}
+
+		// BinaryFilters equalities
+		binaryEqualities := p.BinaryFilter.Equalities()
+		for binaryNS := range binaryEqualities.NotEqual {
+			eq := fEqs.binaryEqualities[binaryNS]
+			// NotEqual == 0, so clear n bitmap bit
+			utils.ClearBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.binaryEqualities[binaryNS] = eq
+		}
+		for binaryNS := range binaryEqualities.Equal {
+			eq := fEqs.binaryEqualities[binaryNS]
+			// Equal == 1, so set n bitmap bit
+			utils.SetBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.binaryEqualities[binaryNS] = eq
+		}
+	}
+
+	return nil
+}
+
+func createNewInnerMap(m *bpf.Module, mapName string, mapVersion uint16) (*bpf.BPFMapLow, error) {
+	// use the map prototype to create a new map with the same properties
+	prototypeMap, err := m.GetMap(mapName)
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	info, err := bpf.GetMapInfoByFD(prototypeMap.FileDescriptor())
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	btfFD, err := bpf.GetBTFFDByID(info.BTFID)
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	opts := &bpf.BPFMapCreateOpts{
+		BTFFD:                 uint32(btfFD),
+		BTFKeyTypeID:          info.BTFKeyTypeID,
+		BTFValueTypeID:        info.BTFValueTypeID,
+		BTFVmlinuxValueTypeID: info.BTFVmlinuxValueTypeID,
+		MapFlags:              info.MapFlags,
+		MapExtra:              info.MapExtra,
+		MapIfIndex:            info.IfIndex,
+	}
+	newInnerMap, err := bpf.CreateMap(
+		prototypeMap.Type(),
+		fmt.Sprintf("%s_%d", mapName, mapVersion), // new map name
+		prototypeMap.KeySize(),
+		prototypeMap.ValueSize(),
+		int(prototypeMap.MaxEntries()),
+		opts,
+	)
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	return newInnerMap, nil
+}
+
+func updateOuterMap(m *bpf.Module, mapName string, mapVersion uint16, innerMap *bpf.BPFMapLow) error {
+	outerMap, err := m.GetMap(mapName)
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	u16Key := mapVersion
+	keyPointer := unsafe.Pointer(&u16Key)
+	innerMapFD := uint32(innerMap.FileDescriptor())
+	valuePointer := unsafe.Pointer(&innerMapFD)
+
+	// update version filter map
+	// - key is the map version
+	// - value is the related filter map FD.
+	if err := outerMap.Update(keyPointer, valuePointer); err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	return nil
+}
+
+func (ps *Policies) createNewFilterMapsVersion(bpfModule *bpf.Module) error {
+	mapsNames := map[string]string{ // inner map name: outer map name
+		UIDFilterMap:         UIDFilterMapVersion,
+		PIDFilterMap:         PIDFilterMapVersion,
```

Comment:
Again, this should be describing the outer/inner relation, why you're creating filter maps and what they mean.

Reply:
![image](https://github.com/aquasecurity/tracee/assets/7395852/1429aa9a-358f-4502-844f-e3aca08631a1)

![image](https://github.com/aquasecurity/tracee/assets/7395852/a82d381c-4ef1-4ed8-b34c-fd3ef664124b)

After sometime not visiting the policies filtering logic, I could only understand/remember taking a look at some old notes of my own. This mean we could do the same, but in a structured way as documentation.